From b2635e7f86b3be4c28433477e982666a26aa13e4 Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Wed, 11 Dec 2024 17:26:14 +0100
Subject: [PATCH] dt-bindings: phy: stm32: usb2phy: use once cell clock
 provider

The OHCI clock must be distinguished from other clocks output of
the PHY. It must be kept active during the resume phase, to access
the registers. So adopt one cells for the clock provider for OHCI
controller to select it.

Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Change-Id: Id0bfeeff1b407a445b28896452e63131f715e238
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/424746
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Christian BRUEL <christian.bruel@foss.st.com>
---
 .../devicetree/bindings/phy/st,stm32-usb2phy.yaml         | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

--- a/Documentation/devicetree/bindings/phy/st,stm32-usb2phy.yaml
+++ b/Documentation/devicetree/bindings/phy/st,stm32-usb2phy.yaml
@@ -153,8 +153,10 @@ properties:
     default: 0
 
   "#clock-cells":
-    description: number of clock cells for ck_usb2phyXXXX_48m consumer
-    const: 0
+    const: 1
+    description:
+      One cell corresponding to the requested clock, 0 for the
+      PLL clock output, 1 for the OHCI clock output.
 
 required:
   - compatible
@@ -176,7 +178,7 @@ examples:
     usb2_phy2: usb-phy {
         compatible      = "st,stm32mp25-usb2phy";
         #phy-cells      = <0>;
-        #clock-cells    = <0>;
+        #clock-cells    = <1>;
         st,syscfg       = <&syscfg 0x2800>;
         clocks          = <&rcc CK_KER_USB2PHY2>;
         resets          = <&rcc USB2PHY2_R>;
